src/Analyzers/CSharp/Analyzers/CodeStyle/CSharpAnalyzerOptionsProvider.cs

```diff
@@ -97,7 +97,7 @@ internal CSharpCodeGenerationOptions GetCodeGenerationOptions()
     public CodeStyleOption2<ExpressionBodyPreference> PreferExpressionBodiedLambdas => GetOption(CSharpCodeStyleOptions.PreferExpressionBodiedLambdas, FallbackCodeStyleOptions.PreferExpressionBodiedLambdas);
     public CodeStyleOption2<bool> PreferStaticLocalFunction => GetOption(CSharpCodeStyleOptions.PreferStaticLocalFunction, FallbackCodeStyleOptions.PreferStaticLocalFunction);
 
-    private TValue GetOption<TValue>(Option2<TValue> option, TValue defaultValue)
+    private TValue GetOption<TValue>(SingleValuedOption2<TValue> option, TValue defaultValue)
```

Comment:
Do we need to retain the `2` suffix?

Reply:
Thinking out loud - this renaming actually makes us inconsistent with the corresponding public types. We had `Option` as a public type in Workspaces layer and `Option2` as the corresponding internal type in the shared layer (CodeStyle and Workspaces). Similarly, we had `PerLanguageOption` as a public type in Workspaces layer and `PerLanguageOption2` as the corresponding internal type in the shared layer. I think this similar naming was very helpful in associating them. After this change we have 4 unrelated names: `Option`, `SingleValuedOption2`, `PerLanguageOption` and `PerLanguageOption2`.

@Youssef1313 I am wondering if it is better to avoid renaming the classes and instead just rename the underlying interfaces `ILanguageSpecificOption` and `IPerLanguageOption` for clarity.

Reply:
I think it makes sense to rename the interface only. I'll do that.

Reply:
I pushed a commit and the PR is now much smaller and commit-by-commit review is now less useful. So I'll avoid rewriting history.